auto-value-gson/src/main/java/com/ryanharter/auto/value/gson/AutoValueGsonExtension.java

```diff
@@ -336,20 +336,18 @@ public MethodSpec createWriteMethod(TypeName autoValueClassName,
         .addParameter(annotatedParam)
         .addException(IOException.class);
 
+    writeMethod.beginControlFlow("if ($N == null)", annotatedParam);
+    writeMethod.addStatement("$N.nullValue()", jsonWriter);
+    writeMethod.addStatement("return");
+    writeMethod.endControlFlow();
+
     writeMethod.addStatement("$N.beginObject()", jsonWriter);
     for (Map.Entry<Property, FieldSpec> entry : adapters.entrySet()) {
       Property prop = entry.getKey();
       FieldSpec field = entry.getValue();
 
-      if (prop.nullable()) {
```

Comment:
I considered using `nullValue()` instead of letting the adapter handle it but adapters should handle `null` values according to the documentation:
> Type adapters should be prepared to read null from the stream and write it to the stream. Alternatively, they should use nullSafe() method while registering the type adapter with Gson.